content/docs/05-ai-sdk-ui/02-chatbot.mdx

```diff
@@ -349,3 +349,253 @@ export async function POST(req: Request) {
   //...
 }
 ```
+
+## Attachments (Experimental)
+
+The `useChat` hook supports sending attachments along with a message as well as rendering them on the client. This can be useful for building applications that involve sending images, files, or other media content to the AI provider.
+
+There are two ways to send attachments with a message, either by providing a `FileList` object or a list of URLs to the `handleSubmit` function:
+
+### FileList
+
+By using `FileList`, you can send multiple files as attachments along with a message using the file input element. The `useChat` hook will automatically convert them into data URLs and send them to the AI provider.
+
+> **Note:** Currently, only `image/*` and `text/*` content types get automatically converted into [multi-modal content parts](https://sdk.vercel.ai/docs/foundations/prompts#multi-modal-messages). You will need to handle other content types manually.
+
+```tsx filename="app/page.tsx"
+'use client';
+
+import { getTextFromDataUrl } from '@ai-sdk/ui-utils';
+import { useChat } from 'ai/react';
+import { useRef, useState } from 'react';
+
+export default function Page() {
+  const { messages, input, handleSubmit, handleInputChange, isLoading } =
+    useChat({
+      api: '/api/chat',
+    });
```

Comment:
I usually simplify the examples as much as possible (remove everything that's not relevant to this docs section like classNames, unused feature, etc) and highlight the crucial pieces. might be worth considering here. the reader wants to learn about this particular feature, other information will just distract them

Reply:
cleaned up the examples!